aries_vcx/src/handlers/issuance/issuer/state_machine.rs

```diff
@@ -140,6 +150,14 @@ impl IssuerSM {
 
         for (uid, message) in messages {
             match self.state {
+                IssuerFullState::Initial(_) => {
```

Comment:
> There's 2 way to construct SM in ProposalReceived state (from_proposal(...), Initial -> ProposalReceived), but only 1 way to create SM in OffertSet (from_offer(...) )

There are 2 ways to get into `OfferSet` as well: `from_offer` and `create` + `set_offer`, which corresponds to proposal's `from_proposal` and `Initial -> ProposalReceived` via update state because, as issuer, I _receive_ proposal and I send _offer_.

I don't see much use in `set_proposal` (one might argue it is even dangerous, as it might allow the issuer to modify what's perceived as objectively received proposal).

An argument could be that capturing all possible transitions inside the state machine can make the SM logic "easier to verify". On the other hand, it would actually increase the number of lines and therefore the code complexity in some sense.